lib/decors/method_added.rb

```diff
@@ -15,14 +19,14 @@ def handle_method_addition(clazz, method_name)
                 return if @_ignore_additions || declared_decorators.empty?
                 decorator_class, params, blk = declared_decorators.pop
 
+                undecorated_method = clazz.instance_method(method_name)
+                decorator = METHOD_CALLED_TOO_EARLY_HANDLER
+                clazz.send(:define_method, method_name) { |*args, &block| decorator.call(self, *args, &block) }
```

Comment:
The method has to be created before calling `decorator_class.new`, because we are passing it both the old and new version